### PR TITLE
coverage workflow pre-check

### PR DESCRIPTION
**Pull request by oshi97**
Adds a pre-check before sending coverage to Coveralls. This check will fail the action if coverage
drops by more than 2 percent when compared to a particular branch.
The coverage percent is read from the lcov file generated by jest using lcov-parse. A github action
was created to wrap lcov-parse and do some simple calculation.

Also updates actions/checkout and actions/setup-node to v3, since the only change
for both is making node 16 the default.

I will merge main into v1 afterwards. There doesn't seem to be magic semantic versioning for reusable workflows,
so I didn't plan to make a v1.1 since that would require us to update literally all our repos that use these workflows.
Doing so would defeat a lot of the purpose of these workflows being "reusable".

J=SLAP-2131
TEST=manual

tested using https://github.com/yext/answers-react-components/pull/180
see that I if I cause coverage to drop by more than 2, we don't upload to coveralls and the check fails early

successfully failing test run: https://github.com/yext/answers-react-components/runs/6908712378?check_suite_focus=true

successfully passing test run: https://github.com/yext/answers-react-components/runs/6911292123?check_suite_focus=true

### CODE REVIEW

File: .github/workflows/coverage.yml
Comment (yen-tt) [inline .github/workflows/coverage.yml:64]: can we include which branch correspond to which percentage?

File: .github/workflows/coverage.yml
Comment (oshi97) [reply .github/workflows/coverage.yml:28]: oh are you thinking of running them in parallel?

File: .github/workflows/coverage.yml
Comment (oshi97) [reply .github/workflows/coverage.yml:64]: Are you thinking of just including an echo somewhere? can definitely do that

File: .github/workflows/coverage.yml
Comment (oshi97) [reply .github/workflows/coverage.yml:28]: We only run coveralls once in the end

File: .github/workflows/coverage.yml
Comment (oshi97) [reply .github/workflows/coverage.yml]: tested and we can get rid of it!

File: .github/workflows/coverage.yml
Comment (yen-tt) [reply .github/workflows/coverage.yml:59]: oh I think you are echoing the percentanges here no? yea that would be nice

File: .github/workflows/coverage.yml
Comment (yen-tt) [reply .github/workflows/coverage.yml:28]: yea, just thought it may be good to to run them in parallel so both branch need to install, build, and run `yext/slapshot-reusable-workflows/get-coverage-percent`.

File: .github/workflows/coverage.yml
Comment (oshi97) [reply .github/workflows/coverage.yml]: no this isn't echoing the percentage, it's echoing the percentIsTooLow "boolean"

Comment (oshi97) [reply .github/workflows/coverage.yml]: this is what a run looks like https://github.com/yext/answers-react-components/runs/6908003012?check_suite_focus=true

Comment (yen-tt) [reply .github/workflows/coverage.yml]: sweet!

File: .github/workflows/coverage.yml
Comment (nmanu1) [inline .github/workflows/coverage.yml]: will the branch be updated after merging into main?

File: get-coverage-percent/package.json
Comment (nmanu1) [inline get-coverage-percent/package.json:7]: can this be renamed to third party notices like we have in our other repos?

File: get-coverage-percent/package.json
Comment (nmanu1) [inline get-coverage-percent/package.json:12]: can the Github Action Licenses spreadsheet in shared drive be updated to include this one?

File: .github/workflows/coverage.yml
Comment (oshi97) [reply .github/workflows/coverage.yml]: it's parallelized now! here is the test run https://github.com/yext/answers-react-components/runs/6908712378?check_suite_focus=true

File: .github/workflows/coverage.yml
Comment (nmanu1) [reply .github/workflows/coverage.yml]: just saw that you updated!

File: get-coverage-percent/package.json
Comment (oshi97) [reply get-coverage-percent/package.json:12]: I went to double check that we were using BSD-3 for our other repos, and noticed that answers-headless also says ISC here... probably due to it being the default for npm init. I made an item to make sure our other repos have licenses in sync with their package.json field https://yexttest.atlassian.net/browse/SLAP-2154
headless react doesn't have a license field

Comment (oshi97) [reply get-coverage-percent/package.json:12]: I copied what core has for the license field and also the bsd-3 license. I didn't see a need to add the preamble but I can if you feel like it's needed

File: get-coverage-percent/package.json
Comment (oshi97) [reply get-coverage-percent/package.json:7]: updated! it's still in the dist folder, I can move it one level up if that feels better

File: .github/workflows/coverage.yml
Comment (oshi97) [reply .github/workflows/coverage.yml]: it's updated already!

File: get-coverage-percent/LICENSE
Comment (nmanu1) [inline get-coverage-percent/LICENSE:30]: looks like some extra github stuff here

File: get-coverage-percent/package.json
Comment (nmanu1) [reply get-coverage-percent/package.json]: oops I wasn't clear. I meant that we might want to update [this spreadsheet](https://docs.google.com/spreadsheets/d/1LTZgKwtAYTA6M_g5iZrg5BAmF8bW3RPBzq33MZaLQ20/edit#gid=0)

Comment (oshi97) [reply get-coverage-percent/package.json]: Oh I wasn't sure if ISC was allowed?

Comment (oshi97) [reply get-coverage-percent/package.json]: If it is I can update

Comment (nmanu1) [reply get-coverage-percent/package.json]: I think ISC is only allowed for some use cases, so if we're licensing under BSD-3, that's probably better and we can just list it on that spreadsheet.

(this spreadsheet is basically trying to do what Snyk does and keep track of the licenses of the dependencies in this repo since there isn't a top-level package.json. so whenever a new github action is added in a workflow in this repo, we should be updating the spreadsheet and getting legal's approval if necessary)

Comment (oshi97) [reply get-coverage-percent/package.json]: Oh I thought this was a different spreadsheet

Comment (oshi97) [reply get-coverage-percent/package.json]: Do we need to put our own github actions into this sheet?

Comment (nmanu1) [reply get-coverage-percent/package.json]: yeah, I'm not 100% sure if we need to for our own actions. we could also put the package-lock for get-coverage-percent in Snyk, but not sure if that's necessary either. maybe @tmeyer2115 will have some thoughts on this

Comment (oshi97) [reply get-coverage-percent/package.json]: this is true licenses are an incredibly crucial part of the software development lifecycle

File: .github/workflows/coverage.yml
Comment (oshi97) [reply .github/workflows/coverage.yml]: no get-coverage-percent manually reads the lcov it doesn't send anything to coveralls

File: get-coverage-percent/package.json
Comment (oshi97) [reply get-coverage-percent/package.json]: I will add Snyk after this pr is merged in

File: get-coverage-percent/LICENSE
Comment (nmanu1) [inline get-coverage-percent/LICENSE:3]: nit: change to 2022

Comment (oshi97) [reply get-coverage-percent/LICENSE]: updated! also updated the license item to check these

**Review by nmanu1 — APPROVED** [review]

**Pull request merged**
